test/firebase_auth_mocks_test.dart

```diff
@@ -484,6 +492,18 @@ void main() {
     );
   });
 
+  test('User.linkWithCredential can throw exception', () async {
```

Comment:
You probably don't need `async` here since none of the test set up uses await. Even if the expect awaits a future error, the test will wait for this check to pass.

https://pub.dev/packages/test#future-matchers